test_conformance/math_brute_force/common.cpp

```diff
@@ -530,3 +552,46 @@ __kernel void )", kernel_name.c_str(), R"((__global RETTYPE_SCALAR* out,
 
     return kernel.str();
 }
+
+cl_int BuildKernels(BuildKernelInfo &info, cl_uint job_id,
+                    SourceGenerator generator)
+{
+    // Generate the kernel code.
+    cl_uint vector_size_index = gMinVectorSizeIndex + job_id;
+    auto kernel_name = GetKernelName(vector_size_index);
+    auto source = generator(kernel_name, info.nameInCode, vector_size_index);
+    std::array<const char *, 1> sources{ source.c_str() };
+
+    // Create the program.
+    clProgramWrapper &program = info.programs[vector_size_index];
+    auto options = GetBuildOptions(info.relaxedMode);
+    int error =
+        create_single_kernel_helper(gContext, &program, nullptr, sources.size(),
+                                    sources.data(), nullptr, options.c_str());
+    if (error != CL_SUCCESS)
+    {
+        vlog_error("\t\tFAILED -- Failed to create program. (%d)\n", error);
+        return error;
+    }
+
+    // Create a kernel for each thread. cl_kernels aren't thread safe, so make
+    // one for every thread
+    auto &kernels = info.kernels[vector_size_index];
+    assert(kernels.empty() && "Dirty BuildKernelInfo");
+    kernels.resize(info.threadCount);
+    for (auto &kernel : kernels)
+    {
+        kernel = clCreateKernel(program, kernel_name.c_str(), &error);
+        if (!kernel || error != CL_SUCCESS)
+        {
+            char buffer[2048];
```

Comment:
I know the old code hardcoded `2048`, but we could take this opportunity to make the code slightly better by querying the log size instead. Something like:
```cpp
size_t log_size;
clGetProgramBuildInfo(program, gDevice, CL_PROGRAM_BUILD_LOG, 0, nullptr, &size);
std::vector<char> buffer(log_size);
clGetProgramBuildInfo(program, gDevice, CL_PROGRAM_BUILD_LOG, log_size, buffer.data(), NULL);
```